.github/workflows/daily-sanitizers.yml

```diff
@@ -0,0 +1,92 @@
+name: daily-sanitizers
+
+on:
+  schedule:
+    - cron: '0 4 * * *' # run at 4 AM UTC
+
+jobs:
+  build:
+    runs-on: [self-hosted, linux, ARM64]
+    strategy:
+      matrix:
+        # Test of these containers
+        container: ["ubuntu-dev:22"]
+        build-type: [Debug]
+        compiler: [{ cxx: g++, c: gcc }]
+        cxx_flags: ["-Werror"]
+    timeout-minutes: 45
+    env:
+      SCCACHE_GHA_ENABLED: "true"
+      SCCACHE_CACHE_SIZE: 6G
+      SCCACHE_ERROR_LOG: /tmp/sccache_log.txt
+
+    container:
+      image: ghcr.io/romange/${{ matrix.container }}
+      credentials:
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.4
+
+      - name: Configure Cache Env
+        uses: actions/github-script@v7
+        with:
+          script: |
+            core.exportVariable('ACTIONS_CACHE_URL', process.env.ACTIONS_CACHE_URL || '');
+            core.exportVariable('ACTIONS_RUNTIME_TOKEN', process.env.ACTIONS_RUNTIME_TOKEN || '')
+
+      - name: Prepare Environment
+        run: |
+          uname -a
+          cmake --version
+          mkdir -p ${GITHUB_WORKSPACE}/build
+
+          echo "===================Before freeing up space ============================================"
+          df -h
+          rm -rf /hostroot/usr/share/dotnet
+          rm -rf /hostroot/usr/local/share/boost
+          rm -rf /hostroot/usr/local/lib/android
+          rm -rf /hostroot/opt/ghc
+          echo "===================After freeing up space ============================================"
+          df -h
+
+      - name: Configure & Build
+        run: |
+          echo "ulimit is"
+          ulimit -s
+          echo "-----------------------------"
+          echo "disk space is:"
+          df -h
+          echo "-----------------------------"
+          mkdir -p $GITHUB_WORKSPACE/build
+          cd $GITHUB_WORKSPACE/build
+          cmake .. \
+            -DCMAKE_BUILD_TYPE=Debug \
+            -GNinja \
+            -DCMAKE_C_COMPILER="${{matrix.compiler.c}}" \
+            -DCMAKE_CXX_COMPILER="${{matrix.compiler.cxx}}" \
+            -DCMAKE_C_COMPILER_LAUNCHER=sccache \
+            -DCMAKE_CXX_COMPILER_LAUNCHER=sccache \
+            -DCMAKE_CXX_FLAGS="${{matrix.cxx_flags}}" \
+            -DWITH_ASAN=ON \
+            -DWITH_USAN=ON
+
+          ninja src/all
+
+      - name: Test
+        run: |
+            cd $GITHUB_WORKSPACE/build
+            ctest -V -L DFLY -E "^(search_family|multi)_test"
```

Comment:
we have 2 test cases that fail right? not the whole search family / multi test
if this is the case I would disable the specif tests that fail
if there is no way to do this with ctest args than add in the test code for the failing tests ifdef WITH_ASAN skip

Reply:
I did. There is a downside to this though. The problem happens on arm only, so the tests become disabled on x86 as well. I don't expect this to be of high friction so it's fine. Besides I do want a long term solution for this as I would like to run the sanitizers as part of every PR.

Reply:
oh I used #ifndef because cmake labels are specific to a test and not a test case :(